lib/project/project-files-manager.ts

```diff
@@ -94,6 +94,14 @@ export class ProjectFilesManager implements Project.IProjectFilesManager {
 		return this.isFileExcluded(relativeToProjectPath, excludedProjectDirsAndFiles, projectDir);
 	}
 
+	public excludeFile(projectDir: string, excludeFilePath: string) : void {
+		if (!this.isProjectFileExcluded(projectDir, excludeFilePath)) {
+			var relativeToProjectPath = path.relative(projectDir, excludeFilePath);
+			var ignoreFilePath = path.join(projectDir, ProjectFilesManager.IGNORE_FILE);
+			this.$fs.appendFile(ignoreFilePath, relativeToProjectPath).wait();
```

Comment:
This would break the .abignore file, if the last existing line in it does not end with a new line char